Assets/Scripts/Runtime/Waves/WaveController.cs

```diff
@@ -0,0 +1,124 @@
+﻿using UnityEngine;
+
+namespace Assets.Scripts.Runtime.Waves
+{
+    [System.Serializable]
+    public class WaveController
+    {
+        [SerializeField] private Transform[] _spawnPoints;
+        [SerializeField] private ScriptableObjects.Stages[] _stages;
+        [SerializeField] private Transform _finalPoint;
```

Comment:
ideally AI would be smart enough to attack player if spawn points are correct
if AIs need more guiding this should generally not be global to all spawn points and enemies

this version is ok as a default to override if needed